tests/integration/test_services_dedup_embedder_extractor_hasher_detector.py

```
@@ -0,0 +1,719 @@
+            hasher.get_file_size(tmp_path / "nope.txt")
+
+    def test_validate_algorithm_valid(self) -> None:
+        from file_organizer.services.deduplication.hasher import FileHasher
+
+        assert FileHasher.validate_algorithm("MD5") == "md5"
+        assert FileHasher.validate_algorithm("SHA256") == "sha256"
+
+    def test_validate_algorithm_invalid(self) -> None:
+        from file_organizer.services.deduplication.hasher import FileHasher
+
+        with pytest.raises(ValueError):
```

Ruff (0.15.9)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)